osqp

hanyujin02 opened the pull request.

hanyujin02 committed:
modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   include/trajectory_planner/mpcPlanner.h

hanyujin02 committed:
modified:   include/trajectory_planner/mpcPlanner.cpp

hanyujin02 committed:
modified:   cfg/mpc_interactive/occupancy_map.yaml
	modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   include/trajectory_planner/mpcPlanner.h
	modified:   src/mpc_node.cpp

hanyujin02 committed:
modified:   cfg/mpc_interactive/obstacle_test_param.yaml
	modified:   cfg/mpc_interactive/occupancy_map.yaml
	modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   include/trajectory_planner/mpcPlanner.h
	modified:   src/mpc_node.cpp

hanyujin02 committed:
modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   src/mpc_node.cpp

hanyujin02 committed:
Update README.md

hanyujin02 committed:
Update README.md

hanyujin02 committed:
modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   rviz/mpc_interactive.rviz

hanyujin02 committed:
Merge branch 'test-osqp' of https://github.com/hanyujin02/trajectory_planner into test-osqp

hanyujin02 committed:
Update README.md

hanyujin02 committed:
modified:   cfg/mpc_interactive/mpc_param.yaml
	modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   include/trajectory_planner/mpcPlanner.h
	modified:   src/mpc_node.cpp

hanyujin02 committed:
Merge branch 'test-osqp' of https://github.com/hanyujin02/trajectory_planner into test-osqp

hanyujin02 committed:
modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   include/trajectory_planner/mpcPlanner.h
	modified:   src/mpc_node.cpp

hanyujin02 committed:
modified:   include/trajectory_planner/mpcPlanner.cpp

hanyujin02 committed:
modified:   include/trajectory_planner/mpcPlanner.cpp

Zhefan-Xu commented on .vscode/settings.json:
这个删了

Zhefan-Xu commented on README.md:
README不要改先

Zhefan-Xu commented on cfg/mpc_interactive/mpc_param.yaml:
using_ACADO和using_OSQP只保留一个，否则两个都是false不好处理，只留一个flag。 或者如果ACADO没有必要可以把相关的全部移除，直接移除这个选项

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.cpp:
```diff
@@ -5,6 +5,7 @@
 */
 
 #include <trajectory_planner/mpcPlanner.h>
+#include <trajectory_planner/third_party/OsqpEigen/OsqpEigen.h>
```
h文件include全都放在.h里，不要放在cpp里

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.cpp:
```diff
@@ -17,6 +18,20 @@ namespace trajPlanner{
 	}
 
 	void mpcPlanner::initParam(){ 
+		if (not this->nh_.getParam(this->ns_ + "/using_ACADO", this->usingACADO_)){
```
同上，只保留一个flag或者移除ACADO

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.cpp:
```diff
@@ -223,18 +238,34 @@ namespace trajPlanner{
 		this->lastRefStartIdx_ = 0;
 	}
 
-	void mpcPlanner::updateDynamicObstacles(const std::vector<Eigen::Vector3d>& obstaclesPos, const std::vector<Eigen::Vector3d>& obstaclesVel, const std::vector<Eigen::Vector3d>& obstaclesSize){
```
这个函数不要修改API，和其他planner保持一致。如果目的是使用其他预测方式，那在class里面添加预测的模块，但是现在这个版本先不要添加，先把osqp相关的迭代一下，其他后面PR再说。

如果你想修改makeplan里面线性部分，那你增加一个linearPredction之类的函数，通过当前obstacle计算，而不是外部传入。

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.cpp:
```diff
@@ -376,8 +407,19 @@ namespace trajPlanner{
 	// }
-
 	bool mpcPlanner::makePlan(){
+		bool retVal;
```
同上，只保留一个flag。或者直接取消ACADO

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.cpp:
```diff
@@ -660,6 +702,39 @@ namespace trajPlanner{
 	}
 
+	std::vector<staticObstacle> mpcPlanner::sortStaticObstacles(const std::vector<staticObstacle> &staticObstacles){
+		if (staticObstacles.size()>21){
```
不要出现21这种hardcode

sort vector本身C++有函数，你这个复杂度太高了，给这个函数加个注释。我的理解是你想要ACADO优先近处

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.cpp:
```diff
@@ -943,12 +1018,12 @@ namespace trajPlanner{
 		    line.lifetime = ros::Duration(0.1);
 
 			for (int i = 0; i<this->dynamicObstaclesPos_.size();i++){
-				double x = this->dynamicObstaclesPos_[i](0); 
```
和前面一样，还是保留之前的写法。

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.cpp:
```diff
@@ -1012,4 +1087,437 @@ namespace trajPlanner{
+	Eigen::SparseMatrix<double> &constraintMatrix, int numObs, int mpcWindow, 
+	std::vector<Eigen::Matrix<double, Eigen::Dynamic, 3>> &oxyz, std::vector<Eigen::Matrix<double, Eigen::Dynamic, 3>> &osize, std::vector<Eigen::Matrix<double, Eigen::Dynamic, 1>> &yaw, 
+	std::vector<std::vector<int>> &isDyamic){
+	// TODO: derive a Eigen::SparseMatrix<double> constraintMatrix
```
TODO看看，确认一下

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.h:
```diff
@@ -26,6 +28,9 @@
 // USING_NAMESPACE_ACADO
 using std::cout; using std::endl;
 namespace trajPlanner{
+	const int numStates = 8;
```
放在class外的理由是什么？

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.h:
```diff
@@ -58,12 +63,15 @@ namespace trajPlanner{
-		std::vector<Eigen::Vector3d> dynamicObstaclesPos_;
-		std::vector<Eigen::Vector3d> dynamicObstaclesVel_;
-		std::vector<Eigen::Vector3d> dynamicObstaclesSize_;
+		std::vector<std::vector<Eigen::Vector3d>> dynamicObstaclesPos_;
```
同上，原来写法

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.h:
```diff
@@ -58,12 +63,15 @@ namespace trajPlanner{
 
 
 		// parameters
+		bool usingACADO_;
```
保留其一。或者去掉ACADO

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.h:
```diff
@@ -96,10 +104,39 @@ namespace trajPlanner{
 		void updatePath(const nav_msgs::Path& path, double ts);
 		void updatePath(const std::vector<Eigen::Vector3d>& path, double ts);
-		void updateDynamicObstacles(const std::vector<Eigen::Vector3d>& obstaclesPos, const std::vector<Eigen::Vector3d>& obstaclesVel, const std::vector<Eigen::Vector3d>& obstaclesSize); // position, velocity, size
+		void updateDynamicObstacles(const std::vector<std::vector<Eigen::Vector3d>>& obstaclesPos, const std::vector<std::vector<Eigen::Vector3d>>& obstaclesVel, const std::vector<std::vector<Eigen::Vector3d>>& obstaclesSize); // position, velocity, size
```
同之前

Zhefan-Xu commented on include/trajectory_planner/mpcPlanner.h:
```diff
@@ -96,10 +104,39 @@ namespace trajPlanner{
 		bool makePlan();
+		bool ACADOSolve();
 		// bool makePlanCG();
+		std::vector<staticObstacle> sortStaticObstacles(const std::vector<staticObstacle> &staticObstacles);
```
你这个函数返回copy of vector，可能会效率低

Zhefan-Xu commented on src/mpc_node.cpp:
把dynamic obstacle加回去

Zhefan-Xu reviewed (commented):
主要是dynamicObstacle那块API不要修改，要做prediction的话用单独模块或者函数，可以再makeplan或者其他地方调用。其他地方一些小细节

hanyujin02 committed:
modified:   cfg/mpc_interactive/mpc_param.yaml
	modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   include/trajectory_planner/mpcPlanner.h
	modified:   src/mpc_node.cpp

hanyujin02 committed:
Update README.md

hanyujin02 committed:
Delete .vscode/settings.json

hanyujin02 committed:
Merge branch 'test-osqp' of https://github.com/hanyujin02/trajectory_planner into test-osqp

hanyujin02 committed:
modified:   cfg/mpc_interactive/mpc_param.yaml
	modified:   include/trajectory_planner/mpcPlanner.cpp
	modified:   include/trajectory_planner/mpcPlanner.h
	modified:   src/mpc_node.cpp

hanyujin02 committed:
Merge branch 'test-osqp' of https://github.com/hanyujin02/trajectory_planner into test-osqp

hanyujin02 committed:
Merge branch 'mpc-dev' into test-osqp
